### PR TITLE
Update go to 1.17 & remove deprecated ioutil

### PR DESCRIPTION
**Pull request by rubensf**

### CODE REVIEW

File: go/pkg/chunker/chunker_test.go [hunk 1]
```diff
@@ -114,15 +113,15 @@ func TestChunkerFromBlob(t *testing.T) {
 }
 
 func TestChunkerFromFile(t *testing.T) {
-	execRoot, err := ioutil.TempDir("", t.Name())
+	execRoot, err := os.MkdirTemp("", t.Name())
```
Comment (atetubou) [inline go/pkg/chunker/chunker_test.go:116]: better to use [t.TempDir](https://pkg.go.dev/testing#T.TempDir)?

File: go/pkg/chunker/chunker_test.go [hunk 2]
```diff
@@ -228,15 +227,15 @@ func TestChunkerFromBlob_Reset(t *testing.T) {
 }
 
 func TestChunkerFromFile_Reset(t *testing.T) {
-	execRoot, err := ioutil.TempDir("", t.Name())
+	execRoot, err := os.MkdirTemp("", t.Name())
```
Comment (atetubou) [inline go/pkg/chunker/chunker_test.go:230]: better to use [t.TempDir](https://pkg.go.dev/testing#T.TempDir)?

File: go/pkg/chunker/chunker_test.go [hunk 3]
```diff
@@ -312,15 +311,15 @@ func TestChunkerErrors_ErrEOF(t *testing.T) {
 
 func TestChunkerResetOptimization_SmallFile(t *testing.T) {
 	// Files smaller than IOBufferSize are loaded into memory once and not re-read on Reset.
-	execRoot, err := ioutil.TempDir("", t.Name())
+	execRoot, err := os.MkdirTemp("", t.Name())
```
Comment (atetubou) [inline go/pkg/chunker/chunker_test.go:314]: better to use [t.TempDir](https://pkg.go.dev/testing#T.TempDir)?

File: go/pkg/chunker/chunker_test.go [hunk 4]
```diff
@@ -357,15 +356,15 @@ func TestChunkerResetOptimization_SmallFile(t *testing.T) {
 func TestChunkerResetOptimization_FullData(t *testing.T) {
 	// After FullData is called once, the file contents will remain loaded into memory and not
 	// re-read on Reset, even if the file is larger than IOBufferSize.
-	execRoot, err := ioutil.TempDir("", t.Name())
+	execRoot, err := os.MkdirTemp("", t.Name())
```
Comment (atetubou) [inline go/pkg/chunker/chunker_test.go:359]: better to use [t.TempDir](https://pkg.go.dev/testing#T.TempDir)?

File: go/pkg/client/cas_test.go [hunk 5]
```diff
@@ -1101,7 +1100,7 @@ func TestDownloadActionOutputs(t *testing.T) {
 			&repb.OutputDirectory{Path: "dir2", TreeDigest: treeADigest.ToProto()},
 		},
 	}
-	execRoot, err := ioutil.TempDir("", "DownloadOuts")
+	execRoot, err := os.MkdirTemp("", "DownloadOuts")
```
Comment (atetubou) [inline go/pkg/client/cas_test.go:1103]: better to use [t.TempDir](https://pkg.go.dev/testing#T.TempDir)?

File: go/pkg/client/cas_test.go [hunk 6]
```diff
@@ -1285,7 +1284,7 @@ func TestDownloadActionOutputsErrors(t *testing.T) {
 	ar := &repb.ActionResult{}
 	ar.OutputFiles = append(ar.OutputFiles, &repb.OutputFile{Path: "foo", Digest: digest.NewFromBlob([]byte("foo")).ToProto()})
 	ar.OutputFiles = append(ar.OutputFiles, &repb.OutputFile{Path: "bar", Digest: digest.NewFromBlob([]byte("bar")).ToProto()})
-	execRoot, err := ioutil.TempDir("", "DownloadOuts")
+	execRoot, err := os.MkdirTemp("", "DownloadOuts")
```
Comment (atetubou) [inline go/pkg/client/cas_test.go:1287]: better to use [t.TempDir](https://pkg.go.dev/testing#T.TempDir)?

File: go/pkg/client/cas_test.go [hunk 7]
```diff
@@ -1406,7 +1405,7 @@ func TestDownloadActionOutputsBatching(t *testing.T) {
 				name := fmt.Sprintf("foo_%s", dg)
 				ar.OutputFiles = append(ar.OutputFiles, &repb.OutputFile{Path: name, Digest: dg.ToProto()})
 			}
-			execRoot, err := ioutil.TempDir("", "DownloadOuts")
+			execRoot, err := os.MkdirTemp("", "DownloadOuts")
```
Comment (atetubou) [inline go/pkg/client/cas_test.go:1408]: better to use [t.TempDir](https://pkg.go.dev/testing#T.TempDir)?

File: go/pkg/client/cas_test.go [hunk 8]
```diff
@@ -1680,7 +1679,7 @@ func TestDownloadFiles(t *testing.T) {
 	fooDigest := fake.Put([]byte("foo"))
 	barDigest := fake.Put([]byte("bar"))
 
-	execRoot, err := ioutil.TempDir("", "DownloadOuts")
+	execRoot, err := os.MkdirTemp("", "DownloadOuts")
```
Comment (atetubou) [inline go/pkg/client/cas_test.go:1682]: better to use [t.TempDir](https://pkg.go.dev/testing#T.TempDir)?

File: go/pkg/client/cas_test.go [hunk 9]
```diff
@@ -1719,7 +1718,7 @@ func TestDownloadFilesCancel(t *testing.T) {
 		uo := uo
 		t.Run(fmt.Sprintf("UnifiedDownloads:%t", uo), func(t *testing.T) {
 			t.Parallel()
-			execRoot, err := ioutil.TempDir("", strings.ReplaceAll(t.Name(), string(filepath.Separator), "_"))
+			execRoot, err := os.MkdirTemp("", strings.ReplaceAll(t.Name(), string(filepath.Separator), "_"))
```
Comment (atetubou) [inline go/pkg/client/cas_test.go:1721]: better to use [t.TempDir](https://pkg.go.dev/testing#T.TempDir)?

File: go/pkg/client/tree_test.go [hunk 10]
```diff
@@ -147,7 +146,7 @@ func (c *callCountingMetadataCache) GetCacheMisses() uint64 {
 
 func TestComputeMerkleTreeRemoteWorkingDir(t *testing.T) {
 	callComputeMerkleTree := func(files, inputs []string, virtualInputs []*command.VirtualInput, localWorkingDir, remoteWorkingDir string) (digest.Digest, map[string]int) {
-		root, err := ioutil.TempDir("", "")
+		root, err := os.MkdirTemp("", "")
```
Comment (atetubou) [inline go/pkg/client/tree_test.go:149]: better to use [t.TempDir](https://pkg.go.dev/testing#T.TempDir)?

**Review by atetubou — CHANGES_REQUESTED** [review]

**Comment by rubensf** [comment]
Changed all uses to `t.TempDir`

**Review by ramymedhat — APPROVED** [review]

File: go/pkg/client/cas_test.go [hunk 11]
```diff
@@ -1101,10 +1100,7 @@ func TestDownloadActionOutputs(t *testing.T) {
 			&repb.OutputDirectory{Path: "dir2", TreeDigest: treeADigest.ToProto()},
 		},
 	}
-	execRoot, err := ioutil.TempDir("", "DownloadOuts")
-	if err != nil {
-		t.Fatalf("failed to make temp dir: %v", err)
-	}
+	execRoot := t.TempDir()
```
Comment (atetubou) [inline go/pkg/client/cas_test.go:1103]: remove L1108 too?

File: go/pkg/fakes/server.go [hunk 12]
```diff
@@ -103,7 +102,7 @@ type TestEnv struct {
 func NewTestEnv(t testing.TB) (*TestEnv, func()) {
 	t.Helper()
 	// Set up temp directory.
-	execRoot, err := ioutil.TempDir("", strings.ReplaceAll(t.Name(), string(filepath.Separator), "_"))
+	execRoot, err := os.MkdirTemp("", strings.ReplaceAll(t.Name(), string(filepath.Separator), "_"))
```
Comment (atetubou) [inline go/pkg/fakes/server.go:105]: why not use t.TempDir here?

File: go/pkg/testutil/testutil.go [hunk 13]
```diff
@@ -12,7 +11,7 @@ func CreateFile(t *testing.T, executable bool, contents string) (string, error)
 	if executable {
 		perm = os.FileMode(0766)
 	}
-	tmpFile, err := ioutil.TempFile(os.TempDir(), "")
+	tmpFile, err := os.CreateTemp(os.TempDir(), "")
```
Comment (atetubou) [inline go/pkg/testutil/testutil.go:14]:
```suggestion
	tmpFile, err := os.CreateTemp(t.TempDir(), "")
```

**Review by atetubou — APPROVED** [review]

File: go/pkg/fakes/server.go (hunk 12 above)
Comment (rubensf) [reply go/pkg/fakes/server.go]: didn't notice it :)

**Comment by rubensf** [comment]
fixed all

**Pull request merged**